GrafeasGroup/tor_ocr
# Dev pm

Relevant issue: https://github.com/GrafeasGroup/tor_ocr/issues/30

## Description:

Acceptance Criteria:
- Blossom's record of a post is marked as OCR attempted if call to OCR service succeeds (regardless of if it found content)
- Blossom's record of a post is not marked as OCR attempted if a call to OCR service fails
Updated flow:
- take the ID that we get from Redis when u/ToR puts in a new post to create an OCR attempt from
- ask Blossom for data at the submission endpoint for this ID
- verify that has_ocr_transcription is False
- create and post Transcription object to Blossom for the above submission ID

## Screenshots:
example transcription record:
<img width="678" alt="Screen Shot 2020-01-27 at 10 51 11 PM" src="https://user-images.githubusercontent.com/8950759/73234332-91111f00-4157-11ea-8a3d-32ca8840dfdd.png">


## Testing Instructions:

- start a local instance of blossom api
  - make sure there is a submission at submission/1 as this is the hard-coded submission for testing
- start a redis server
- start the tor_ocr bot with `tor-apprentice
- open redis cli
  - add reddit ids for post to transcribe and place to post transcription
  - set [post to transcribe] [place to post transcription]
  - e.g. `set t3_eqhr0b t3_eqlhj4`
  - now push these ids to the ocr_ids key
  - e.g. `rpush ocr_ids t3_eqhr0b`

tor_ocr bot should see the new post and get an ocr transcription for it. It will then check the api for the submission and then check if the submission's 'has_ocr_transcription' field is false. If it is false, a new transcription record will be created.

## Checklist:
